hazelcast-jet-core/src/main/java/com/hazelcast/jet/Traverser.java

```diff
@@ -125,6 +125,27 @@ public T next() {
     }
 
     /**
+     * Returns a traverser that will emit the same items as this traverser and
+     * will additionally run the supplied action first time this traverser
+     * returns {@code null}.
+     */
+    @Nonnull
+    default Traverser<T> onFirstNull(@Nonnull Runnable action) {
+        return new Traverser<T>() {
+            private Runnable action2 = action;
```

Comment:
Using a `boolean actionRan` is better because the anonymous class already captures the `action` parameter as a final field; `action2` just duplicates it and nulling it out has no effect.